### PR TITLE
[receiver/apache] send port as a resource attribute

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
**Description:** 
Metrics sent by `apachereceiver` now use `apache.server.port` resource attribute. This feature is hidden behind a feature gate.

**Link to tracking Issue:** #14791

**Testing:** 
Unit tests for the config and scraper.

**Documentation:** 
mdatagen

### CODE REVIEW

File: receiver/apachereceiver/config.go [hunk 1]
```diff
@@ -55,5 +59,16 @@ func (cfg *Config) Validate() error {
...
 
 	cfg.serverName = u.Hostname()
+	cfg.port = u.Port()
+
+	if cfg.port == "" {
+		if u.Scheme == "https" {
+			cfg.port = httpsDefaultPort
+		} else if u.Scheme == "http" {
+			cfg.port = httpDefaultPort
+		}
```
Comment (djaglowski) [inline receiver/apachereceiver/config.go:69]: The `Validate` function should not change the configuration.

I suggest moving the `port` field to the scraper and setting it in the factory.

**Comment by aboguszewski-sumo** [comment]
I found an easy way to check the port in the tests anyway, so I modified it.

File: receiver/apachereceiver/config.go (hunk 1 above)
Comment (aboguszewski-sumo) [reply receiver/apachereceiver/config.go:69]: I understand the reason, but `serverName` is set in this function anyway - should it also be changed?

Comment (djaglowski) [reply receiver/apachereceiver/config.go:69]: Yeah, really it should. I think it's an inconsequential change but is better for the sake of consistency, so it's fine to do in this PR in my opinion.

File: receiver/apachereceiver/scraper_test.go [hunk 2]
```diff
@@ -56,7 +56,12 @@ func TestScraper(t *testing.T) {
 	expectedFile := filepath.Join("testdata", "scraper", "expected.json")
 	expectedMetrics, err := golden.ReadMetrics(expectedFile)
 	require.NoError(t, err)
+	url, err := url.Parse(apacheMock.URL)
+	require.NoError(t, err)
+	expectedMetrics.ResourceMetrics().At(0).Resource().Attributes().Remove("apache.server.port")
```
Comment (TylerHelmuth) [inline receiver/apachereceiver/scraper_test.go:61]: Do we need to do this remove before the Put?

File: receiver/apachereceiver/config.go (hunk 1 above)
Comment (aboguszewski-sumo) [reply receiver/apachereceiver/config.go]: I think we have caught a bug here.
After the fix, the integration tests stopped passing - the expected value of `serverName` was `""`, while the receiver returned `"localhost"`. I checked immediately - and it happens that `Validate` is called only inside unit tests - so outside the tests, `serverName` wasn't actually set and was always an empty string. 
I fixed this and pushed the changes on this branch.

File: receiver/apachereceiver/scraper_test.go (hunk 2 above)
Comment (aboguszewski-sumo) [reply receiver/apachereceiver/scraper_test.go]: Apparently we don't. I removed this line.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**